File: agent/library/src/main/java/org/matilda/di/LoggerModule.java
package org.matilda.di;

import dagger.Module;
import dagger.Provides;
import org.matilda.MatildaConnection;
import org.matilda.logger.CompoundLogger;
import org.matilda.logger.Logger;

import java.io.InputStream;
import java.io.OutputStream;

@Module
public class LoggerModule {
    private final Logger mLogger;

    public LoggerModule(Logger... loggers) {
        mLogger = new CompoundLogger(loggers);
    }

    @Provides
    Logger logger() {
        return mLogger;
    }
}

File: agent/library/src/main/java/org/matilda/di/LoggerModule.kt
package org.matilda.di

import dagger.Module
import dagger.Provides
import org.matilda.logger.CompoundLogger
import org.matilda.logger.Logger

@Module
class LoggerModule(vararg loggers: Logger) {
    private val mLogger = CompoundLogger(*loggers)


    @Provides
    fun logger(): Logger {
        return mLogger
    }
}
